src/PositionManager.sol

```diff
@@ -92,22 +94,22 @@ contract PositionManager is
         return _getLocker();
     }
 
-    /// @param params is an encoding of uint256 tokenId, PositionConfig memory config, uint256 liquidity, bytes hookData
+    /// @param params is an encoding of uint256 tokenId, PositionConfig calldata config, uint256 liquidity, bytes hookData
     /// @dev Calling increase with 0 liquidity will credit the caller with any underlying fees of the position
-    function _increase(bytes memory params) internal {
-        (uint256 tokenId, PositionConfig memory config, uint256 liquidity, bytes memory hookData) =
-            abi.decode(params, (uint256, PositionConfig, uint256, bytes));
+    function _increase(bytes calldata params) internal {
+        (uint256 tokenId, PositionConfig calldata config, uint256 liquidity, bytes calldata hookData) =
```

Comment:
why is PositionConfig still calldata? it gets translated to memory when we do toId() no? So I feel like this is doing more work than it needs to?

Reply:
it does but it means its only brought to memory once instead of twice - i've optimised `toId` too though